src/main/java/spark/parser/Parser.java

```diff
@@ -0,0 +1,173 @@
+package spark.parser;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import spark.exceptions.SparkException;
+import spark.parser.commands.CommandKeyword;
+import spark.exceptions.formatexceptions.*;
+import spark.parser.commands.commandtypes.*;
+import spark.parser.params.AddDeadlineParams;
+import spark.parser.params.AddEventParams;
+import spark.parser.params.AddTodoParams;
+
+/**
+ * Holds methods for interpreting user's input.
+ */
+public class Parser {
+    /** Specifies the accepted input-format for dates and times */
+    private static final DateTimeFormatter inputDateTimeFormatter = DateTimeFormatter.ofPattern("M-d-yyyy Hmm");
+
+    /**
+     * Returns a Command specific to the type of operation
+     * that the user wishes to perform.
+     *
+     * @param input what the user has typed into the console
+     * @return      a Command
+     */
+    public static Command parseInput(String input) throws SparkException {
+        String[] tokens = input.split(" "); // split command into individual keywords by single-space
+        String commandKeyword = tokens[0]; // assume that the first keyword is always the command word
+        CommandKeyword keyword = CommandKeyword.getCommand(commandKeyword);
+
+        if (keyword == CommandKeyword.BYE) {
+            return new ExitCommand();
+
+        } else if (keyword == CommandKeyword.LIST) {
+            return new ListCommand();
+
+        } else if (keyword == CommandKeyword.MARK) {
+            return new MarkCommand(getMarkParams(input, keyword));
```

Comment:
I like how you gave each command a different parsing method, it really made the code look much cleaner! 💯